src/System.Threading.Tasks.Extensions/tests/AsyncValueTaskMethodBuilderTests.cs

```diff
@@ -87,7 +156,38 @@ public void Start_InvokesMoveNext()
         [InlineData(2, false)]
         [InlineData(1, true)]
         [InlineData(2, true)]
-        public void AwaitOnCompleted_ForcesTaskCreation(int numAwaits, bool awaitUnsafe)
+        public void NonGeneric_AwaitOnCompleted_ForcesTaskCreation(int numAwaits, bool awaitUnsafe)
+        {
+            AsyncValueTaskMethodBuilder b = default;
+
+            var dsm = new DelegateStateMachine();
+            TaskAwaiter<int> t = new TaskCompletionSource<int>().Task.GetAwaiter();
+
+            Assert.InRange(numAwaits, 1, int.MaxValue);
```

Comment:
What's the point of this check?

Reply:
This is copy-and-paste of the generic one, which is from a while ago.  I assume I was thinking I wanted to ensure that the input test data made sense, so that if the test then failed, it wasn't because of bad inputs.